src/components/Article.jsx

```diff
@@ -1,14 +1,21 @@
 import { Link, useParams } from "react-router-dom";
-import { useState, useEffect } from "react";
-import { getArticle, getComments } from "../../api";
+import { useState, useEffect, useRef } from "react";
+import { getArticle, getComments, postVote } from "../../api";
 import Loading from "./Loading";
 import Aside from "./Aside";
 
 const Article = () => {
   const { article_id } = useParams();
+  const storedVote = JSON.parse(localStorage.getItem(`hasVoted${article_id}`));
   const [article, setArticle] = useState([]);
+  const [votePlus, setVotePlus] = useState(1);
+  const [voteMinus, setVoteMinus] = useState(-1);
+  const [optimisticVote, setOptimisticVote] = useState(0);
   const [comments, setComments] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [renderIgnore, setRenderIgnore] = useState(false);
```

Comment:
I feel there is _alot_ of state here, an it might not all be necessary? Currently votePlus and voteMinus are dependant on eachother, so that logic could be derived from the same piece of state.